### DISCUSSION TITLE
HEIC decoding is missing, even if supported by the environment

### DISCUSSION THREAD

**Discussion by mqudsi** [MicrosoftEdge/WebView2Feedback#3891] [Q&A]
**Description**

An `<img />` node pointing to an HEIC-encoded image is not supported, even if [the relevant Windows plugin](https://www.microsoft.com/store/productId/9PMMSR1CGPWG) is installed.

Maybe this is by design, but since `Windows.Media.Imaging` can decode it, perhaps WebView2 can, too.

**Version**
SDK: Reunion 0.8
Runtime: 91.0.864.67
Framework: Win32
OS: Windows 10 18363

**Comment by champnic** [comment]
@mqudsi Thanks for the bug report! What is the behavior in the Edge browser, is it supported? We generally match the behavior of the Edge browser (and Chromium).

If the image is sent from a server you may be able to use either the `WebResourceRequested` or `WebResourceResponseReceived` events to get the image in your host app and do the decoding yourself with Windows.Media.Imaging. If the image is local, you could do that decode first before having the web load it.

Thanks!

**Comment by champnic** [comment]
If you'd like the web platform to support this feature you can open an issue on https://crbug.com for Chromium or use the feedback tool in Edge (shift + alt + i).

**Comment by mqudsi** [comment]
Thanks for explaining the rationale behind the supported codecs. That makes perfect sense and my request doesn’t fit in under the context of a Edgium or Chromium bug/feature request; as it is behavior expected in shell integrations, not in standalone web browsers.
